sdks/java/io/snowflake/src/main/java/org/apache/beam/sdk/io/snowflake/services/SnowflakeServiceImpl.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.snowflake.services;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.function.Consumer;
+import javax.sql.DataSource;
+import org.apache.beam.sdk.io.snowflake.Location;
+import org.apache.beam.sdk.io.snowflake.SnowflakeCloudProvider;
+import org.apache.beam.sdk.io.snowflake.enums.WriteDisposition;
+import org.apache.beam.sdk.transforms.SerializableFunction;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Implemenation of {@link SnowflakeService} used in production. */
+public class SnowflakeServiceImpl implements SnowflakeService<SnowflakeServiceConfig> {
```

Comment:
Was this file moved with git mv? This appears to be a rename (io/snowflake/SnowflakeServiceImpl.java → io/snowflake/services/SnowflakeServiceImpl.java) followed by additional modifications. It'd be nice for git to be aware of the move explicitly so we could see what the changes are to the file after the move.

This goes for any other files that might have been moved and edited afterwards as well. The SnowflakeService.java rename is an example of what I'd like to see all the renames look like.

Reply:
`SnowflakeServiceImpl.java` is the only file with this kind situation. I tried  
manually to do it using `git mv` but without success  as this file contains more than 50% changes.  (https://stackoverflow.com/questions/54085112/git-mv-and-changes-and-similarity-index)

Reply:
Can we make one commit with move and other with changes? maybe this will work?

Reply:
I just tried it,  it doesn't work :(